--- a/net/minecraft/world/SpawnerAnimals.java
+++ b/net/minecraft/world/SpawnerAnimals.java
@@ -19,8 +19,9 @@
 
 public final class SpawnerAnimals
 {
-    private static final int MOB_COUNT_DIV = (int)Math.pow(17.0D, 2.0D);
+    public static final int MOB_COUNT_DIV = (int)Math.pow(17.0D, 2.0D); // disrespecTech - private -> public
     private final Set<ChunkCoordIntPair> eligibleChunksForSpawning = Sets.<ChunkCoordIntPair>newHashSet();
+    public int lastEligibleChunkCount = 1; // disrespecTech - Mobcap
 
     /**
      * adds all chunks within the spawn radius of the players to eligibleChunksForSpawning. pars: the world,
@@ -165,6 +166,7 @@
                 }
             }
 
+            this.lastEligibleChunkCount = i; // disrespecTech - Mobcap
             return i4;
         }
     }
