--- a/net/minecraft/network/play/server/S47PacketPlayerListHeaderFooter.java
+++ b/net/minecraft/network/play/server/S47PacketPlayerListHeaderFooter.java
@@ -45,4 +45,24 @@
     {
         handler.handlePlayerListHeaderFooter(this);
     }
+    // disrespecTech Start - TPS/MSPT
+
+    public static S47PacketPlayerListHeaderFooter createHeaderOnly(IChatComponent header) {
+        S47PacketPlayerListHeaderFooter packet = new S47PacketPlayerListHeaderFooter(header);
+        packet.footer = new net.minecraft.util.ChatComponentText("");
+        return packet;
+    }
+
+    public static S47PacketPlayerListHeaderFooter createFooterOnly(IChatComponent footer) {
+        S47PacketPlayerListHeaderFooter packet = new S47PacketPlayerListHeaderFooter(new net.minecraft.util.ChatComponentText(""));
+        packet.footer = footer;
+        return packet;
+    }
+
+    public static S47PacketPlayerListHeaderFooter create(IChatComponent header, IChatComponent footer) {
+        S47PacketPlayerListHeaderFooter packet = new S47PacketPlayerListHeaderFooter(header);
+        packet.footer = footer;
+        return packet;
+    }
+    // disrespecTech End
 }
