--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -87,6 +87,7 @@
     /** Instance of Minecraft Server. */
     private static MinecraftServer mcServer;
     private final ISaveFormat anvilConverterForAnvilFile;
+    public final me.totorewa.dissserver.DissServer dissServer; // disrespecTech
 
     /** The PlayerUsageSnooper instance. */
     private final PlayerUsageSnooper usageSnooper = new PlayerUsageSnooper("server", this, getCurrentTimeMillis());
@@ -199,6 +200,7 @@
         this.authService = new YggdrasilAuthenticationService(proxy, UUID.randomUUID().toString());
         this.sessionService = this.authService.createMinecraftSessionService();
         this.profileRepo = this.authService.createProfileRepository();
+        this.dissServer = new me.totorewa.dissserver.DissServer(this); // disrespecTech
     }
 
     protected ServerCommandManager createNewCommandManager()
@@ -516,6 +518,7 @@
                 this.statusResponse.setServerDescription(new ChatComponentText(this.motd));
                 this.statusResponse.setProtocolVersionInfo(new ServerStatusResponse.MinecraftProtocolVersionIdentifier("1.8.8", 47));
                 this.addFaviconToStatusResponse(this.statusResponse);
+                this.dissServer.onServerStart(); // disrespecTech
 
                 while (this.serverRunning)
                 {
@@ -669,6 +672,7 @@
         }
 
         this.theProfiler.startSection("root");
+        this.dissServer.startTick(this.tickCounter); // disrespecTech
         this.updateTimeLightAndEntities();
 
         if (i - this.nanoTimeSinceStatusRefresh >= 5000000000L)
@@ -698,6 +702,7 @@
         this.theProfiler.startSection("tallying");
         this.tickTimeArray[this.tickCounter % 100] = System.nanoTime() - i;
         this.theProfiler.endSection();
+        this.dissServer.endTick(this.tickCounter); // disrespecTech
         this.theProfiler.startSection("snooper");
 
         if (!this.usageSnooper.isSnooperRunning() && this.tickCounter > 100)
